builtin/logical/ad/path_config.go

```diff
@@ -0,0 +1,219 @@
+package ad
+
+import (
+	"context"
+
+	"github.com/fatih/structs"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/helper/activedirectory"
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+const (
+	// This length is arbitrarily chosen but should work for
+	// most Active Directory minimum and maximum length settings.
+	// A bit of tongue-in-cheek since programmers love their base-2 exponents.
+	defaultPasswordLength = 64
+
+	// The number of minutes in 32 days.
+	defaultPasswordTTLs = 24 * 60 * 32
+)
+
+type configHandler struct {
+	logger hclog.Logger
+}
+
+func (h *configHandler) Path() *framework.Path {
+	return &framework.Path{
+		Pattern: "config",
+		Fields: map[string]*framework.FieldSchema{
+			"username": {
+				Type:        framework.TypeString,
+				Default:     "",
```

Comment:
We don't need to specify an "" default value. Same comment for all the rest of the TypeString fields.

Reply:
Nice! Thanks, will clean all that up.